lib/Dancer2/Serializer/Mutable.pm

```diff
@@ -31,25 +25,44 @@ my $serializer = {
     },
 };
 
-sub support_content_type {
-    my ( $self, $ct ) = @_;
-
-    # FIXME: are we getting full content type?
+has mapping => (
+    is   => 'ro',
+    lazy => 1,
+    default => sub {
+        my $self = shift;
+
+        if ( my $mapping = $self->config->{mapping} ) {
+
+            # initialize non-default serializers
+            for my $s ( values %$mapping ) {
+                # TODO allow for arguments via the config
+                next if $serializer->{$s};
+                my $serializer_object = ('Dancer2::Serializer::'.$s)->new;
+                $serializer->{$s} = {
+                    from => sub { shift; $serializer_object->deserialize(@_) },
+                    to   => sub { shift; $serializer_object->serialize(@_)   },
+                };
+            }
 
-    if ( $ct && grep +( $_ eq $ct ), keys %{$formats} ) {
-        $self->set_content_type($ct);
+            return $mapping;
+        }
 
-        return 1;
-    }
 
-    return 0;
-}
+        return {
+            'text/x-yaml'        => 'YAML',
+            'text/html'          => 'YAML',
+            'text/x-data-dumper' => 'Dumper',
```

Comment:
I do wonder whether Dumper should be in the default mapping at all - I imagine its use in real-world Dancer apps is pretty unlikely, and IIRC there's some potential security issues with deserializing untrusted data with it (although Dancer2::Serializer::Dumper does use Safe.pm to do the eval in a compartment, but it still makes me nervous).  I think I'd be in favour of just dropping it - it's easy enough to add back in if required!